File: The Best Box.py
    import math
     
    t = int(input())
    for i in range(t):
        p, s = map(int, input().split())
        p = float(p)
        s = float(s)
        d = math.sqrt(((p/6)*(p/6))-((2*s)/3))
        b = ((p/6) - d)/2
        v = (b * b * b) - ((p/4) * b * b) + ((s/2) * b)
        print(round(v,2)) 

File: The Best Box.cpp
    #include <stdio.h>
    #include <math.h>
    int main ()
     {
        int t,p,s;
        float v,x;
        scanf ("%u", &t);
        for(;t>0;t--)
         {
            scanf ("%u %u", &p,&s);
            x = (float)p/12 - sqrt(p*p-24*s)/12;
            v = x*(x*x-(x*p/4)+s/2);
            printf ("%.2f\n", v);
         }
        return 0;
     }
